parquet/src/main/java/com/netflix/iceberg/parquet/ParquetMetricsRowGroupFilter.java

```diff
@@ -156,6 +157,13 @@ public Boolean or(Boolean leftResult, Boolean rightResult) {
       Preconditions.checkNotNull(struct.field(id),
           "Cannot filter by nested column: %s", schema.findField(id));
 
+      // When filtering nested types notNull() is implicit filter passed even
```

Comment:
So Spark will take a filter like `a.b = 12` and attempt to push down `notNull(a)`. Is that correct?

Because Parquet only has stats for primitive columns, the value count is null and this assumes that the column is missing when it is actually just not a leaf of the schema.

I agree with this fix. Are there other cases that also need to be supported for correctness? It seems like this should be checked in every visitor method (`isNull`, `equal`, `notEqual`, etc.) for correctness.

An alternative is to also check the predicate on all child types, but I think this simple solution is fine for now. Maybe we should note in this comment that we could do more work to check?

Reply:
Re: Spark  passing down notNull(a) , yes, It's an implicit filter passed down when one uses nested types in filters.

Re: other such cases. 
Equals case might be affected, will add the fix to that case as well. The isNull(col) doesn't need fixing as it returns ROWS_MIGHT_MATCH when valueCounts is checked. The notEqual case returns ROWS_MIGHT_MATCH by default so this might not apply there.